src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -4,1096 +4,686 @@
+// The SettingsLoader class is an internal implementation detail of CascadiaSettings.
+// Member methods aren't safe against abuse and you need to ensure to call them in a specific order.
+// See CascadiaSettings::LoadAll() for instance.
+SettingsLoader::SettingsLoader(const std::string_view& userJSON, const std::string_view& inboxJSON)
 {
-    std::string msg;
+    _parse(OriginTag::InBox, {}, inboxJSON, inboxSettings);
 
     try
     {
-        throw;
+        _parse(OriginTag::User, {}, userJSON, userSettings);
     }
     catch (const JsonUtils::DeserializationError& e)
     {
-        static constexpr std::string_view basicHeader{ "* Line {line}, Column {column}\n{message}" };
-        static constexpr std::string_view keyedHeader{ "* Line {line}, Column {column} ({key})\n{message}" };
+        _rethrowSerializationExceptionWithLocationInfo(e, userJSON);
+    }
 
-        std::string jsonValueAsString{ "array or object" };
-        try
+    if (const auto sources = userSettings.globals->DisabledProfileSources())
+    {
+        _ignoredNamespaces.reserve(sources.Size());
+        for (const auto& id : sources)
         {
-            jsonValueAsString = e.jsonValue.asString();
-            if (e.jsonValue.isString())
-            {
-                jsonValueAsString = fmt::format("\"{}\"", jsonValueAsString);
-            }
+            _ignoredNamespaces.emplace(id);
         }
-        catch (...)
+    }
+
+    // See member description of _userProfileCount.
+    _userProfileCount = userSettings.profiles.size();
+}
+
+// Generate dynamic profiles and add them to the list of "inbox" profiles
+// (meaning profiles specified by the application rather by the user).
+void SettingsLoader::GenerateProfiles()
+{
+    const auto executeGenerator = [&](const auto& generator) {
+        const auto generatorNamespace = generator.GetNamespace();
+
+        if (!_ignoredNamespaces.count(generatorNamespace))
         {
-            // discard: we're in the middle of error handling
+            try
+            {
+                generator.GenerateProfiles(inboxSettings.profiles);
+            }
+            CATCH_LOG_MSG("Dynamic Profile Namespace: \"%.*s\"", gsl::narrow<int>(generatorNamespace.size()), generatorNamespace.data())
         }
+    };
 
-        msg = fmt::format("  Have: {}\n  Expected: {}", jsonValueAsString, e.expectedType);
-
-        auto [l, c] = _LineAndColumnFromPosition(settingsString, e.jsonValue.getOffsetStart());
-        msg = fmt::format((e.key ? keyedHeader : basicHeader),
-                          fmt::arg("line", l),
-                          fmt::arg("column", c),
-                          fmt::arg("key", e.key.value_or("")),
-                          fmt::arg("message", msg));
-        throw SettingsTypedDeserializationException{ msg };
-    }
+    executeGenerator(PowershellCoreProfileGenerator{});
+    executeGenerator(WslDistroGenerator{});
+    executeGenerator(AzureCloudShellGenerator{});

```

Comment:
will you handle the merge conflicts for the visual studio profile generator if you merge first?

Reply:
I'd prefer merging the other PR first and handling the conflicts here.